packages/@vuepress/core/lib/node/loadTheme.js

```diff
@@ -60,8 +60,15 @@ module.exports = function loadTheme (ctx) {
 }
 
 function normalizeThemePath (resolved) {
-  const { entry, fromDep } = resolved
+  const { entry, name, fromDep } = resolved
   if (fromDep) {
+    const packageRoot = require.resolve(`${name}/package.json`)
+    const { main } = require(packageRoot)
+    if (main.endsWith('.vue')) {
```

Comment:
Could this break if `main` field is not defined in `package.json` ?

Reply:
Uh you are right. I just reverted this code snippet from #1564

Reply:
Tweaked